typescript/bots/ts_libs/bots.d.ts

```diff
@@ -22,9 +22,14 @@ export interface HeroMatchup {
     counter: string[];
 }
 
-
 export type BotRole = "pos_1" | "pos_2" | "pos_3" | "pos_4" | "pos_5";
 export type ItemBuilds = { [key in BotRole]: string[] };
 export type SkillBuilds = { [key in BotRole]: number[] };
 export type TalentBuilds = { [key in BotRole]: TalentTreeBuild };
 export type HeroMatchups = { [key: string]: HeroMatchup };
+
+export interface GameState {
+    defendPings: {
+        pingedTime: number;
+    } | null;
+}
```

Comment:
lets order this file in the way `interface`s are on top and `type`s on bottom

Reply:
I don't think that would make a lot of sense from organizational perspective since it could make different definitions scattered across the file, I'd say it's better to logically group them together, e.g. interfaces `BotSetup`, `TalentTreeBuild` and types `BotRole`, `ItemBuilds`, `SkillBuilds`, `TalentBuilds` could be grouped together since they're all related to bot script setup, while `HeroMatchup` and `HeroMatchups` don't have anything to do with them.
Generally I try to group things like that not by type but by their purpose/domain area.

Reply:
the code base doesnt need to grow big for the bots. HeroMatchup may be more related to hero selection not individual bot setup yes, but we don't need to put those in a separate file. I dont think managing a lot of really small file is a good choice when there is a purpose to add an interface.

So we can think of this bot.d file as a generic file for holding all the interfaces we want to add for the ts scripts here. We dont need to separate those until it actual grow too big to effectively maintain in a file.

Since we put them in a single file, ordering them is going to make them easier to manage. Putting similar purpose stuff all together might be good if there's only a few purposes. But as the amount purposes grow, it will look messy since the length of the type of the properties are different. And different purposes might just stick all together since they can all be related in some ways, e.g. all related to HeroInternalName (to be added later), Role, etc, then the purpose will be vague and won't make much sense since the types are meant to be for multi-purposes.

We have good IDE, it's convenient to find any types we want especially we have them in a file.